umpleonline/ump/GeometricSystem.ump

```diff
@@ -115,143 +115,154 @@ class NonPolyhedra{
 class Circle{
 	isA CurvedShape;
 	double radius;
-	double getArea(){/*implementation */}
-	double getPerimeterLength(){/*implementation */}
-	boolean isLessThan(CurvedShape object){/*implementation */}
-	boolean isEqual(CurvedShape object){/*implementation */}
+	double getArea(){return 1.0;}
+	double getPerimeterLength(){return 1.0;}
+	boolean isLessThan(CurvedShape object){return true;}
+	boolean isEqual(CurvedShape object){return true;}
 }
 class Ellipse {
 	isA CurvedShape;
 	double majorAxis;
 	double minorAxis;
 	Point fFocus;
 	Point gFocus;
-	Point[2] getFoci(){/*implementation */}
-	double getArea(){/*implementation */}
-	double getPerimeterLength(){/*implementation */}
-	Line getTangent(){/*implementation */}
-	boolean isLessThan(CurvedShape object){}
-	boolean isEqual(CurvedShape object){}
+	Point[] getFoci(){return new Point[2];}
+	double getArea(){return 1.0;}
+	double getPerimeterLength(){return 1.0;}
+	Line getTangent(){return new Line(fFocus, gFocus);}
+	boolean isLessThan(CurvedShape object){return true;}
+	boolean isEqual(CurvedShape object){return true;}
 }
 class Quadrilateral{
 	isA Polygon;
-	double getArea(){/*implementation */}
-	double getPerimeterLength(){/*implementation */}
-	boolean isLessThan(Polygon object){}
-	boolean isEqual(Polygon object){}
+	double getArea(){return 1.0;}
+	double getPerimeterLength(){return 1.0;}
+	boolean isLessThan(Polygon object){return true;}
+	boolean isEqual(Polygon object){return true;}
+	boolean isEqual(CurvedShape object){return false;}
 }
 
 class Rectangle{
 	isA Quadrilateral;
 	//overwrite for better calculation
-	double getArea(){/*implementation */}
+	double getArea(){return 1.0;}
 	//overwrite for better calculation
-	double getPerimeterLength(){/*implementation */}	
+	double getPerimeterLength(){return 1.0;}	
 }
 class ArbitraryPolygon{
 	isA Polygon;
-	double getArea(){/*implementation */}
-	double getPerimeterLength(){/*implementation */}
-	boolean isLessThan(Polygon object){}
-	boolean isEqual(Polygon object){}	
+	double getArea(){return 1.0;}
+	double getPerimeterLength(){return 1.0;}
+	boolean isLessThan(Polygon object){return true;}
+	boolean isEqual(Polygon object){return true;}
+	boolean isEqual(CurvedShape object){return true;}	
```

Comment:
Looks like the following method should be deleted.
`boolean isEqual(CurvedShape object){return false;}`